### PR TITLE
Buildhooks patch 1

### PR DESCRIPTION
**Pull request by abanuog**
Added the build hooks modules, and patches (including one custom patch). Updated the
composer.lock file as well.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -102,6 +104,12 @@
         "patches": {
             "drupal/ldap":{
                 "Fix LDAP Help Redirect Link": "https://www.drupal.org/files/issues/2019-09-03/3076155-ldap-help-redirect-4.patch"
+            },
+            "drupal/build_hooks":{
+                "Response event should send frontend environment entity instead of plugin": "https://www.drupal.org/files/issues/2019-06-03/3059157-build_hooks-event-2.patch" 
+            },
+            "drupal/build_hooks_azure":{
+                "Our patch for build request format": "https://raw.githubusercontent.com/ccswbs/drupal-8-composer/buildhooks-patch-1/bha_id.patch"
```
Comment (brharp) [inline composer.json]: Does this hard code the name of the branch?

Comment (abanuog) [reply composer.json]: created a gist, and updated the patch URL

Comment (brharp) [reply composer.json]: We can delete bha_id.patch from the repository then.

**Review by brharp — APPROVED** [review]

**Review by mmafe — APPROVED** [review]
Looks good.

**Pull request merged**